python/ctsm/site_and_regional/base_case.py

```diff
@@ -0,0 +1,213 @@
+
+    Methods
+    -------
+    create_1d_coord(filename, lon_varname , lat_varname,x_dim , y_dim )
+        create 1d coordinate variables to enable sel() method
+
+    add_tag_to_filename(filename, tag)
+       add a tag and timetag to a filename ending with
+       [._]cYYMMDD.nc or [._]YYMMDD.nc
+
+    update_metadata(nc)
+        Class method for adding some new attributes (such as date, username) and
+        remove the old attributes from the netcdf file.
+    """
+
+    def __init__(self, create_domain, create_surfdata, create_landuse, create_datm):
+        """
+        Initializes BaseCase with the given arguments.
+
+        Parameters
+        ----------
+        create_domain : bool
+            Flag for creating domain file a region/single point
+        create_surfdata : bool
+            Flag for creating domain file a region/single point
+        create_landuse : bool
+            Flag for creating landuse file a region/single point
+        create_datmdata : bool
+            Flag for creating datm files a region/single point
+        """
+        self.create_domain = create_domain
+        self.create_surfdata = create_surfdata
+        self.create_landuse = create_landuse
+        self.create_datm = create_datm
+
+    def __str__(self):
+        """
+        Converts ingredients of the BaseCase to string for printing.
+        """
+        return (
+            str(self.__class__)
+            + "\n"
+            + "\n".join(
+                (
+                    str(item) + " = " + str(self.__dict__[item])
+                    for item in sorted(self.__dict__)
+                )
+            )
+        )
+
+    @staticmethod
+    def create_1d_coord(filename, lon_varname, lat_varname, x_dim, y_dim):
+        """
+        Create 1d coordinate variables for a netcdf file to enable sel() method
+
+        Parameters
+        ----------
+            filename (str) : name of the netcdf file
+            lon_varname (str) : variable name that has 2d lon
+            lat_varname (str) : variable name that has 2d lat
+            x_dim (str) : dimension name in X -- lon
+            y_dim (str): dimension name in Y -- lat
+
+        Raises
+        ------
+            None
+
+        Returns
+        -------
+            f_out (xarray Dataset): Xarray Dataset with 1-d coords
+
+        """
+        logging.debug("Open file: " + filename)
+        f_in = xr.open_dataset(filename)
+
+        # create 1d coordinate variables to enable sel() method
+        lon0 = np.asarray(f_in[lon_varname][0, :])
+        lat0 = np.asarray(f_in[lat_varname][:, 0])
+        lon = xr.DataArray(lon0, name="lon", dims=x_dim, coords={x_dim: lon0})
+        lat = xr.DataArray(lat0, name="lat", dims=y_dim, coords={y_dim: lat0})
+
+        f_out = f_in.assign({"lon": lon, "lat": lat})
+
+        f_out.reset_coords([lon_varname, lat_varname])
+        f_in.close()
+        return f_out
+
+    @staticmethod
+    def add_tag_to_filename(filename, tag):
```

Comment:
The function Sam added was directly copied from this PR and these lines in subset_data : https://github.com/negin513/ctsm/blob/b854ad6b0b3e393cf4f1ab100edd3d085ef0cd14/tools/contrib/subset_data.py#L445-L#461. 
The original function adapted from Sean Swenson's code... 
I wish the PR #1448 was not done in this way so that the history was preserved for git blame purposes...:wq

Your issue is addressed. I removed this from base_case.py...